### DISCUSSION TITLE
Matomo migration to Plausible? Possible?

### DISCUSSION THREAD

**Discussion by berkes** [plausible/analytics#2160] [Anything Else]
I'm considering moving from my self-hosted matomo to either the hosted matomo(paid), or the hosted (paid) Plausible. The TCO is just getting too high: the server alone is almost on par with the monthly fee for Plausible.

I like both the product and pricing of Plausible better. 

In order to migrate from matomo to Plausible I see several options:

* Export from matomo, [convert], then import that in Plausible. History retained.
* Just archive all matomo sites, take the server offline, and make a clean start in Plausible. No history.
* Run matomo alongside of Plausible for months or years until the "history" in Matomo is irrelevant (we hardly look back years)
* Migrate self-hosted matomo to hosted matomo. This option is offered.

I'd prefer the first option. But so far, I see the `[convert]` part missing. Nor do I see a way to avoid that step and import a matomo-export directly. I'd expect I need to convert the matomo-export (CSVs) into the format that Google Analytics exports. As an import of that GA-export format is supported by Plausible.

Has anyone ever done this? Is a matomo-export/import on the roadmap for Plausible? Are there scripts, or articles out there that help or describe a migration from matomo to plausible?

**Comment by metmarkosaric** [comment]
thanks for your interest in Plausible, @berkes! we don't have any import from Matomo unfortunately. there's only the GA(UA) import for now. could be nice to build a more generic import tool that works for various providers in the future but there are no short term plans to do that. we are a small team with limited resources so need to focus on more important / popular requests first. there's a thread that we will update with our progress: #211

**Comment by culturerich** [comment]
Just to add, I appreciate it's not a priority, but I would also be interested in this migration: Matomo > Plausible.

**Comment by ampaze** [comment]
Maybe this might help, thats the way I did it:

https://gist.github.com/ampaze/3a22d3c2adbc1d0bbeff409966ccd132

**Comment by metmarkosaric** [comment]
**Update:** You can now import external data into Plausible using CSV files. The CSV import supports Plausible data by default but can also be used to import data from other sources if the external data is formatted according to our guidelines. See the full details: https://plausible.io/docs/csv-import
